### PR TITLE
Fix #782: Use PasswordProvider for the HashiCorp Vault Token

### LINKED ISSUE #782: Securely handle the HashiCorp Vault Token in Kroxylicious configuration
When connecting to HashiCorp Vault, the client must pass a Vault Token (a secret) in order to authenticate.  Currently the token is obtained from the Kroxylicious configuration where it is stored in-line as a string.  This approach would raise red flags during a security audit.   We need a secure mechanism(s) to inject the secret.

In a Kubernetes environment, utilising a Kubernetes Secret to is an obvious choice.

### PR DESCRIPTION
**Pull request by k-wall**
### Type of change

_Select the type of your PR_

- Enhancement / new feature

### Description

Fix #782.

why: Security audits are likely to flag the fact that the password is inline as a fail. Now we have the PasswordProvider object as part of the API, this filter can take advantage of it too.)

### Additional Context

_Why are you making this pull request?_

### Checklist

_Please go through this checklist and make sure all applicable tasks have been done_

- [x] Write tests
- [x] Make sure all tests pass
- [ ] Review performance test results. Ensure that any degradations to performance numbers are understood and justified.
- [ ] Make sure all Sonar-Lint warnings are addressed or are justifiably ignored.
- [x] Update documentation
- [x] Reference relevant issue(s) and close them after merging
- [x] For user facing changes, update CHANGELOG.md (remember to include changes affecting the API of the test artefacts too).

### CODE REVIEW

File: kroxylicious-kms-provider-hashicorp-vault/src/main/java/io/kroxylicious/kms/provider/hashicorp/vault/config/Config.java [hunk 1]
```diff
@@ -14,18 +14,19 @@
 
 import com.fasterxml.jackson.annotation.JsonProperty;
 
+import io.kroxylicious.proxy.config.tls.PasswordProvider;
```
Comment (robobario) [inline kroxylicious-kms-provider-hashicorp-vault/src/main/java/io/kroxylicious/kms/provider/hashicorp/vault/config/Config.java:17]: would be good to move this to a different package, doesn't need to be part of this PR

**Review by robobario — APPROVED** [review]
LGTM

**Review by SamBarker — APPROVED** [review]
LGTM

**Comment by k-wall** [comment]
I'll move the classes as part of this PR.  I'm thinking of using package name '.secret'

File: kroxylicious-kms-provider-hashicorp-vault/src/main/java/io/kroxylicious/kms/provider/hashicorp/vault/config/Config.java (hunk 1 above)
Comment (k-wall) [reply kroxylicious-kms-provider-hashicorp-vault/src/main/java/io/kroxylicious/kms/provider/hashicorp/vault/config/Config.java]: Done, I moved them to '.secret'.

**Comment by k-wall** [comment]
@robobario please re-review.  It is just the package name and a couple of sonar fixes (no functional changes intended). Happy for you to merge it if you are happy.

**Review by robobario — APPROVED** [review]

**Pull request merged**